src/main/java/org/nuxeo/segment/io/SegmentIOComponent.java

```diff
@@ -185,97 +181,102 @@ public void identify(NuxeoPrincipal principal) {
+                gtraits.putAll(groupMeta);
+                group((String) groupMeta.get("id"), wrapper.getUserId(), gtraits, options);
             } else {
-                log.debug("send identify with " + metadata.toString());
-                Traits traits = new Traits();
-                traits.putAll(wrapper.getMetadata());
-                Context ctx = new Context();
-                ctx.setProviders(getProviders());
-                Analytics.identify(wrapper.getUserId(), traits, ctx);
-
-                Map<String, Serializable> groupMeta = wrapper.getGroupMetadata();
-                if (groupMeta.size() > 0 && groupMeta.containsKey("id")) {
-                    Traits gtraits = new Traits();
-                    gtraits.putAll(groupMeta);
-                    group((String) groupMeta.get("id"), wrapper.getUserId(),
-                            gtraits, ctx);
-                } else {
-                    // automatic grouping
-                    if (principal.getCompany() != null) {
-                        group(principal.getCompany(), wrapper.getUserId(),
-                                null, ctx);
-                    } else if (wrapper.getMetadata().get("company") != null) {
-                        group((String) wrapper.getMetadata().get("company"),
-                                wrapper.getUserId(), null, ctx);
-                    }
+                // automatic grouping
+                if (principal.getCompany() != null) {
+                    group(principal.getCompany(), wrapper.getUserId(), null, options);
+                } else if (wrapper.getMetadata().get("company") != null) {
+                    group((String) wrapper.getMetadata().get("company"), wrapper.getUserId(), null, options);
                 }
             }
         }
     }
 
-    protected void group(String groupId, String userId, Traits traits,
-            Context ctx) {
+    protected void group(String groupId, String userId, Traits traits, Options options) {
         if (groupId == null || groupId.isEmpty()) {
             return;
         }
-        Flusher flusher = getFlusher();
-        if (flusher != null) {
-            Group grp = new Group(userId, groupId, traits, new DateTime(), ctx);
-            flusher.enqueue(grp);
+
+        if (Framework.isTestModeSet()) {
+            pushForTest("group", userId, traits, options);
         } else {
-            log.warn("Can not use Group API");
+            Flusher flusher = getFlusher();
+            if (flusher != null) {
+                Group grp = new Group(userId, groupId, traits, options);
+                flusher.enqueue(grp);
+            } else {
+                log.warn("Can not use Group API");
+            }
         }
     }
 
-    protected void pushForTest(String action, String principalName,
-            String eventName, Map<String, Serializable> metadata) {
+    protected void pushForTest(String action, String userId, Traits traits, Options options) {
```

Comment:
Yeah, I agree but it would mean mockups in the tests and... maybe for the next time ? ;)